### ISSUE TITLE
Java14 enhanced instanceof check validation: UnusedImportsCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8379]
Child of #8391 
Check documentation: https://checkstyle.sourceforge.io/config_imports.html#UnusedImports
Identified at https://github.com/checkstyle/checkstyle/pull/8294#issuecomment-650896023

From check documentation:
_Checks for unused import statements_

This check should not apply to pattern variables.

**Comment by nmancus1** [comment]
@esilkensen @pbludov since we've introduced the new PATTERN_VARIABLE_DEF token, we can close this one too, right?

**Comment by esilkensen** [comment]
Agreed, just confirming that if the import is used only in an instanceof pattern this check detects that and doesn’t flag as a violation?

**Comment by nmancus1** [comment]
>Agreed, just confirming that if the import is used only in an instanceof pattern this check detects that and doesn’t flag as a violation?

```
➜  instanceof-pattern-var-def /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.

➜  instanceof-pattern-var-def cat TestClass.java
import java.util.List;

public class TestClass {
    static{
        Object obj = "test";
        if (obj instanceof List list) {
        }
    }
}%                                                                              

➜  instanceof-pattern-var-def cat config.xml    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="UnusedImports"/>
  </module>
</module>

➜  instanceof-pattern-var-def java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
Audit done.

```

**Comment by romani** [comment]
we need two tests: with violation and without violation ALWAYS.

@esilkensen , no violations mean - Check does not work.

**Comment by esilkensen** [comment]
> we need two tests: with violation and without violation ALWAYS.
> 
> 
> 
> @esilkensen , no violations mean - Check does not work.

I don’t understand - there shouldn’t be any violations in that example because the import is used by the instanceof expression?

**Comment by nmancus1** [comment]
With violation:
```
➜  instanceof-pattern-var-def /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  instanceof-pattern-var-def cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="UnusedImports"/>
  </module>
</module>
➜  instanceof-pattern-var-def cat TestClass.java 
import java.util.List;


public class TestClass {
    static{
        Object obj = "test";
        if (obj instanceof String str) {
        }
    }
}%                                                                              ➜  instanceof-pattern-var-def java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Error: Unable to access jarfile /home/nick/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar
➜  instanceof-pattern-var-def java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/instanceof-pattern-var-def/TestClass.java:1:8: Unused import - java.util.List. [UnusedImports]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
we process IDENT https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java#L231

Pattern do have IDENT under TYPE - https://github.com/checkstyle/checkstyle/pull/8401/files#diff-d9a48e7d8e8fa192ed23a1af1597506dR3561 , so it works .

no changes are required.
